helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java

```diff
@@ -922,25 +922,26 @@ public HostStatus call() throws Exception {
 
   protected HostStatus awaitHostStatusWithLabels(final HeliosClient client,
                                                  final String host,
-                                                 final HostStatus.Status status)
+                                                 final HostStatus.Status status,
+                                                 final Map<String, String> labels)
       throws Exception {
-    return awaitHostStatusWithLabels(client, host, status, LONG_WAIT_SECONDS, SECONDS);
-  }
 
-  protected HostStatus awaitHostStatusWithLabels(final HeliosClient client, final String host,
-                                                 final HostStatus.Status status,
-                                                 final int timeout,
-                                                 final TimeUnit timeUnit) throws Exception {
-    return Polling.await(timeout, timeUnit, new Callable<HostStatus>() {
-      @Override
-      public HostStatus call() throws Exception {
-        final HostStatus hostStatus = getOrNull(client.hostStatus(host));
-        if (hostStatus == null || hostStatus.getLabels().size() == 0) {
-          return null;
-        }
-        return (hostStatus.getStatus() == status) ? hostStatus : null;
+    final HostStatus hostStatus = Polling.await(LONG_WAIT_SECONDS, SECONDS, () -> {
+      final HostStatus candidate = getOrNull(client.hostStatus(host));
+
+      if (candidate == null || candidate.getStatus() != status
+          // labels are stored in ZK after the host has come up
+          || candidate.getLabels().size() == 0) {
+
+        return null;
       }
+      return candidate;
     });
+
+    assertThat("host " + host + " has status=" + status + " with labels=" + hostStatus.getLabels(),
+        hostStatus.getLabels(), is(labels));
```

Comment:
moved the assertion into this method rather than assert something like this in each callsite as it was previously